lib/internal/repl.js

```diff
@@ -119,7 +119,7 @@ function setupHistory(repl, historyPath, oldHistoryPath, ready) {
     }
 
     if (data) {
-      repl.history = data.split(/[\n\r]+/, repl.historySize);
+      repl.history = data.split(/[\n\r]+/).slice(repl.historySize * -1);
```

Comment:
I doubt it would be an issue large enough to necessitate that unless some manually edited in a way-too-big file.